File: solutions/2356-largest-combination-with-bitwise-and-greater-than-zero/largest-combination-with-bitwise-and-greater-than-zero.py
class Solution:
    def largestCombination(self, candidates: List[int]) -> int:
        max_count = 0
        
        # Check each bit position from 0 to 31
        for bit in range(32):
            current_count = 0
            # Count numbers that have 1 at current bit position
            for num in candidates:
                if num & (1 << bit):
                    current_count += 1
            max_count = max(max_count, current_count)
            
        return max_count

File: solutions/2356-largest-combination-with-bitwise-and-greater-than-zero/largest-combination-with-bitwise-and-greater-than-zero.cs
public class Solution {
    private const int BitMax = 24;
    
    public int LargestCombination(int[] candidates) {
        int maxSet = 1;
        int mask = 1;
        
        for (int shift = 0; shift < BitMax; shift++) {
            int count = 0;
            foreach (int num in candidates) {
                if ((num & mask) != 0) {
                    count++;
                }
            }
            maxSet = Math.Max(maxSet, count);
            mask <<= 1;
        }
        return maxSet;
    }
}